hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/S3EventsHoodieIncrSource.java

```diff
@@ -174,7 +178,27 @@ public S3EventsHoodieIncrSource(
     }
     Option<Dataset<Row>> dataset = Option.empty();
     if (!cloudFiles.isEmpty()) {
+      Log.info("Number of files to be processed=" + cloudFiles.size());
       dataset = Option.of(sparkSession.read().format(fileFormat).load(cloudFiles.toArray(new String[0])));
+      if (props.containsKey(HoodieIncrSource.Config.DROP_COLUMNS)) {
+        String dropColumns = props.getString(HoodieIncrSource.Config.DROP_COLUMNS);
+        dataset = dataset.map(ds -> ds.drop(dropColumns.split(",")));
+      }
+      if (props.containsKey(HoodieIncrSource.Config.CAST_TO_STRING)) {
```

Comment:
in general this should go in as a general feature support to any detlastreamer sources. Lets think about what it takes to add such feature. If its not feasible, we can add it just incr source.

Reply:
Okay